Add force-merge API supports primary_only parameter (#6664)
opensearch-project/documentation-website: _api-reference/index-apis/force-merge.md

| `ignore_unavailable` | Boolean | If `true`, OpenSearch ignores missing or closed indexes. If `false`, OpenSearch returns an error if the force merge operation encounters missing or closed indexes. Default is `false`. |
| `max_num_segments` | Integer | The number of larger segments into which smaller segments are merged. Set this parameter to `1` to merge all segments into one segment. The default behavior is to perform the merge as necessary. |
| `only_expunge_deletes` | Boolean | If `true`, the merge operation only expunges segments containing a certain percentage of deleted documents. The percentage is 10% by default and is configurable in the `index.merge.policy.expunge_deletes_allowed` setting. Prior to OpenSearch 2.12, `only_expunge_deletes` ignored the `index.merge.policy.max_merged_segment` setting. Starting with OpenSearch 2.12, using `only_expunge_deletes` does not produce segments larger than `index.merge.policy.max_merged_segment` (by default, 5 GB). For more information, see [Deleted documents](#deleted-documents). Default is `false`. |

> | `primary_only` | Boolean | If set to `true`, the merge operation is performed only on the primary shards of an index. This can be useful when you want to take snapshot of the index after the merge is complete. Snapshots only copy segments from the primary shards. Merging the primary shards can reduce resource consumption. Default is `false`. |

Reviewer: ```suggestion
| `primary_only` | Boolean | If set to `true`, then the merge operation is performed only on the primary shards of an index. This can be useful when you want to take a snapshot of the index after the merge is complete. Snapshots only copy segments from the primary shards. Merging the primary shards can reduce resource consumption. Default is `false`. |
```

> | `primary_only` | Boolean | If `true`, the merge operation only performs on the primary shards, this is useful when you want to take snapshot for the specified indexes(snapshot only copies segments from the primary shards) after the merge operation completes, performing the merge operation only on primary shards can reduce the resources consumption. Default is `false`. |

Reviewer 2: ```suggestion
| `primary_only` | Boolean | If set to `true`, the merge operation is performed only on the primary shards of an index. this can be useful when you want to take snapshot of the index after the merge is complete. Snapshots only copy segments from the primary shards. Merging the primary shards can reduce resource consumption. Default is `false`. |
```

```
{% include copy-curl.html %}

> #### Example request: Perform force merge operation only on the primary shards of a index

Reviewer 2: ```suggestion
#### Example request: Force merge primary shards
```
